Improve test coverage for tools.

GPHemsley opened the pull request:
Improves test coverage for tools (particularly tools/lib/template_parser.py) by:
* Adding tests to cover untested lines.
* Removing the tests directory from the coverage calculation. (Testing the tests seems redundant.)

This PR also adds an exclusion line to the naïve pattern for checking for missing tuples in format strings, to keep the linter happy.

smarx commented:
*Automated message from Dropbox CLA bot*

@GPHemsley, it looks like you've already signed the Dropbox CLA. Thanks!

GPHemsley commented:
I made some additional changes:
* Replaced generic `Exception` with `TemplateParserException`.
* Restored assertion that `state.depth` is 0, since no code path could result in anything else.
* Fixed `IndexError` when there is only zero or more whitespace characters between `<` and `>`. (`str.split()` will return an empty list in this case, which means there is no index 0.)

GPHemsley commented:
I restored the check of `state.depth`, which is non-zero if there are start tags without end tags, and included a test for it.

showell commented:
Looks good so far.  Can you rebase these commits into two commits:

- exclude tests from coverage reports
- add test coverage and fix small bugs

If it's not too hard to break out a commit that just addresses the IndexError bug, I guess that would be nice to be separate from a commit integrity standpoint.

More details on our commit conventions are here: http://zulip.readthedocs.io/en/latest/version-control.html


-

GPHemsley commented:
Changes made and force-pushed. I also factored out the null token replacement change.

GPHemsley commented:
Cleaned up commit message in 4th commit.
